# Exploratory data analysis of internet access

This notebook examines how internet access has evolved globally from 2000 to 2025, with a focus on the gap between income groups and world regions and whether that gap is closing. I chose 2000 as it the point where data becomes more reliable as prior to 2000, data coverage of low-income countries is quite sparse.

## Data Source

- [Our World in Data – Share of the population using the Internet](https://ourworldindata.org/grapher/share-of-individuals-using-the-internet?country=WB_SA~WB_NA~WB_SSA~WB_EAP)

## Objectives

- How has global internet adoption evolved between 2000 and 2025?
- What are the internet adoption differences between socioeconomic groups?
- What are the internet adoption differences between regions?
- Is the adoption gap narrowing between regions and socioeconomic groups?
- What was the state of global internet adoption in 2025?


### Setup & Imports

In [140]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import plotly.express as px
import ipywidgets as widgets
from IPython.display import display

sns.set_style('whitegrid')

In [141]:
data = pd.read_csv("../data/raw/share-of-individuals-using-the-internet.csv")

### Goals

#### 1. Global internet adoption trend. How has the share of people using the internet changed worldwide between 2000 and 2025?

In [142]:
data_2000_2025 = data[data['Year'] >= 2000]
data_2000_2025

,Entity,Code,Year,Share of the population using the Internet
1,Afghanistan,AFG,2001,0.004723
2,Afghanistan,AFG,2002,0.004561
3,Afghanistan,AFG,2003,0.087891
4,Afghanistan,AFG,2004,0.105809
5,Afghanistan,AFG,2005,1.224150
...,...,...,...,...
6326,Zimbabwe,ZWE,2019,26.588300
6327,Zimbabwe,ZWE,2020,29.298600
6328,Zimbabwe,ZWE,2021,37.096700
6329,Zimbabwe,ZWE,2022,37.886100


In [143]:
world = data_2000_2025[data_2000_2025['Entity'] == 'World']

In [144]:
fig = px.line(
    world,
    x='Year',
    y='Share of the population using the Internet',
    title='Global growth on share of people using the Internet',
    labels={'Share of the population using the Internet': 'Share of people using the Internet (%)'},
    markers=True
)
fig.update_layout(
    xaxis=dict(showgrid=True, gridcolor='lightgrey', gridwidth=1),
    yaxis=dict(showgrid=True, gridcolor='lightgrey', gridwidth=1),
    plot_bgcolor='white'
)
fig.write_image('../visualizations/global-growth.png')
fig.show()

Based on the line graph plot above, it is clear that global internet adoption has grown steadily over time. The World trend shows an increase in the share of people using the Internet grew by 58 percentage points, from ~15.6% in 2000 to ~73.6% in 2025. The line is roughly linear which means that there has been no significant change in the rate of growth over time around 2-3% every year.

#### 2. Internet adoption by socioeconomic group. How does adoption differ across income groups, and has the gap between them narrowed?

In [145]:
socioeconomic = data_2000_2025[data_2000_2025['Entity'].str.contains('income')]['Entity'].unique()
group_data = data_2000_2025[data_2000_2025['Entity'].isin(socioeconomic)]

In [146]:
fig = px.line(
    group_data,
    x='Year',
    y='Share of the population using the Internet',
    color='Entity',
    category_orders={'Entity': [
        'High-income countries',
        'Upper-middle-income countries',
        'Lower-middle-income countries',
        'Low-income countries'
    ]},
    title='Internet adoption by socioeconomic group',
    labels={'Share of the population using the Internet': 'Share of people using the Internet (%)', 'Entity': 'Income group'},
    markers=True
)
fig.update_layout(
    xaxis=dict(showgrid=True, gridcolor='lightgrey', gridwidth=1),
    yaxis=dict(showgrid=True, gridcolor='lightgrey', gridwidth=1),
    plot_bgcolor='white'
)
fig.write_image('../visualizations/internet-adoption-by-socioeconomic-group.png')
fig.show()

The gap between high-income and low-income countries in 2005 was 51.9 percentage points, low-income countries had 1% of share of people using the internet and high-income countries had 52.9% share of people using the internet. Two decades later the gap has increased to 70.8 percentage points, low-income countries have 23.4% of share of people using the internet and high-income countries have 94.2% share of people using the internet. Upper-middle-income countries adoption of internet grew the fastest. The gap between high-income countries and upper-middle-income countries is almost closed, but the gap between the other countries have increased, likely due to political instabilities, lack of accessible and affordable infrastructure for the general public or different governmental priorities.

#### 3. Internet adoption by region. How does adoption differ across world regions, and has the gap narrowed?

In [147]:
world_regions = data_2000_2025[data_2000_2025['Entity'].str.contains('WB')]['Entity'].unique()
world_regions_data = data_2000_2025[data_2000_2025['Entity'].isin(world_regions)]
world_regions_data

,Entity,Code,Year,Share of the population using the Internet
1588,East Asia and Pacific (WB),WB_EAP,2005,14.7
1589,East Asia and Pacific (WB),WB_EAP,2006,16.6
1590,East Asia and Pacific (WB),WB_EAP,2007,20.8
1591,East Asia and Pacific (WB),WB_EAP,2008,25.2
1592,East Asia and Pacific (WB),WB_EAP,2009,29.5
...,...,...,...,...
5429,Sub-Saharan Africa (WB),WB_SSA,2021,29.0
5430,Sub-Saharan Africa (WB),WB_SSA,2022,31.1
5431,Sub-Saharan Africa (WB),WB_SSA,2023,32.3
5432,Sub-Saharan Africa (WB),WB_SSA,2024,33.6


In [148]:
world_regions_data = world_regions_data.copy()
world_regions_data['Entity'] = world_regions_data['Entity'].str.replace(' (WB)', '', regex=False)

fig = px.line(
    world_regions_data,
    x='Year',
    y='Share of the population using the Internet',
    color='Entity',
    category_orders={'Entity': [
        'North America',
        'Europe and Central Asia',
        'East Asia and Pacific',
        'Latin America and Caribbean',
        'South Asia',
        'Middle East and North Africa, Afghanistan and Pakistan',
        'Sub-Saharan Africa'
    ]},
    title='Internet adoption by world region',
    labels={'Share of the population using the Internet': 'Share of people using the Internet (%)', 'Entity': 'World region'},
    markers=True
)
fig.update_layout(
    xaxis=dict(showgrid=True, gridcolor='lightgrey', gridwidth=1),
    yaxis=dict(showgrid=True, gridcolor='lightgrey', gridwidth=1),
    plot_bgcolor='white'
)
fig.write_image('../visualizations/internet-adoption-by-world-region.png')
fig.show()

In 2025 the highest adoption is North America with 95.6% and the lowest is Sub-Saharan Africa with 35.7%. South Asia grew the most from 2.1% in 2005 to 67.8% in 2025. The gap is narrowing as the gap between top and bottom region in 2005 was 66.3 percentage points and in 2025 it is 59.9 percentage points. However the growth of 6.4 percentage points over 20 years is quite low.

#### 4. Is the adoption gap narrowing between regions and socioeconomic groups?

In [149]:
gap_income = group_data.pivot(index='Year', columns='Entity', values='Share of the population using the Internet')
gap_income['gap'] = gap_income['High-income countries'] - gap_income['Low-income countries']

In [150]:
fig = px.line(
    gap_income,
    x=gap_income.index,
    y='gap',
    title='Gap in Internet adoption between high-income and low-income countries',
    labels={'gap': 'Gap in Internet adoption (%)', 'Year': 'Year'},
    markers=True
)
fig.update_layout(
    xaxis=dict(showgrid=True, gridcolor='lightgrey', gridwidth=1),
    yaxis=dict(showgrid=True, gridcolor='lightgrey', gridwidth=1),
    plot_bgcolor='white'
)
fig.write_image('../visualizations/gap-in-internet-adoption.png')
fig.show()

In [151]:
gap_region = world_regions_data.pivot(index='Year', columns='Entity', values='Share of the population using the Internet')
gap_region['gap'] = gap_region['North America'] - gap_region['Sub-Saharan Africa']

In [152]:
fig = px.line(
    gap_region,
    x=gap_region.index,
    y='gap',
    title='Gap in Internet adoption between North America and Sub-Saharan Africa',
    labels={'gap': 'Gap in Internet adoption (%)', 'Year': 'Year'},
    markers=True
)
fig.update_layout(
    xaxis=dict(showgrid=True, gridcolor='lightgrey', gridwidth=1),
    yaxis=dict(showgrid=True, gridcolor='lightgrey', gridwidth=1),
    plot_bgcolor='white'
)
fig.write_image('../visualizations/gap-in-internet-adoption-region.png')
fig.show()

In [153]:
gap_combined = pd.DataFrame({
    'Year': gap_income.index,
    'Income gap (High vs Low)': gap_income['gap'].values,
    'Regional gap (North America vs Sub-Saharan Africa)': gap_region['gap'].values
})

In [154]:
fig = px.line(
    gap_combined,
    x='Year',
    y=['Income gap (High vs Low)', 'Regional gap (North America vs Sub-Saharan Africa)'],
    labels={'value': 'Gap in Internet adoption (%)', 'variable': 'Gap type'},
    category_orders={'variable': [
        'Income gap (High vs Low)',
        'Regional gap (North America vs Sub-Saharan Africa)'
    ]},
    title='Gap in Internet adoption between high-income vs low-income countries and North America vs Sub-Saharan Africa',
    markers=True
)
fig.update_layout(
    xaxis=dict(showgrid=True, gridcolor='lightgrey', gridwidth=1),
    yaxis=dict(showgrid=True, gridcolor='lightgrey', gridwidth=1),
    plot_bgcolor='white'
)
fig.write_image('../visualizations/gap-in-internet-adoption-combined.png')
fig.show()

The two gaps are moving in different directions. The income gap has plateaued after increasing, while the regional gap is gradually narrowing. In 2025, the income gap stands at 70.8 percentage points compared to 59.9 percentage points for the regional gap, suggesting that digital inequality is more strongly tied to income than geography. Lower-income countries may be prioritizing more pressing needs such as security, healthcare, or political stability over internet infrastructure investment.

#### 5. What was the state of global internet adoption in 2023?

Switched to 2023 from 2025 as 2023 is the most recent year with comprehensive country-level data.


In [155]:
aggregates = list(world_regions) + list(socioeconomic) + ['World']
data_2023 = data_2000_2025[(data_2000_2025['Year'] == 2023) & (~data_2000_2025['Entity'].isin(aggregates))]

In [156]:
top10 = data_2023.nlargest(10, 'Share of the population using the Internet')
bottom10 = data_2023.nsmallest(10, 'Share of the population using the Internet')
top10['Group'] = 'Top 10'
bottom10['Group'] = 'Bottom 10'

In [157]:
fig = px.bar(
    pd.concat([top10, bottom10]).sort_values('Share of the population using the Internet', ascending=True),
    x='Share of the population using the Internet',
    y='Entity',
    color='Group',
    title='Top 10 and Bottom 10 countries by share of people using the Internet in 2023',
    labels={'Share of the population using the Internet': 'Share of people using the Internet (%)', 'Entity': 'Country'},
)

fig.update_layout(
    xaxis=dict(showgrid=True, gridcolor='lightgrey', gridwidth=1),
    yaxis=dict(showgrid=True, gridcolor='lightgrey', gridwidth=1, tickmode='linear'),
    plot_bgcolor='white'
)

fig.write_image('../visualizations/top-bottom-10.png')
fig.show()

In [158]:
print(top10['Share of the population using the Internet'].mean())

99.57614


In [159]:
print(bottom10['Share of the population using the Internet'].mean())

17.30895


The top 10 countries are primarily high-income nations from the Global North, while the bottom 10 are largely low-income countries from the Global South and Sub-Saharan Africa, consistent with the findings of the previous analysis. The disparity in internet access between these groups is substantial: the top 10 countries have an average internet usage rate of 99.6%, compared to just 17.3% in the bottom 10 countries.

This digital divide has important implications. Limited internet access can restrict opportunities for income generation, education, and access to essential services. It may also reinforce existing global inequalities, as emerging technologies, particularly artificial intelligence, are likely to be adopted more rapidly in highly connected, high-income countries. Since access to such technologies depends heavily on internet infrastructure, there is a risk that low-income countries will fall further behind, widening the existing development gap.

### Conclusion